### ISSUE TITLE
Loading of background layers

### ISSUE THREAD

**Issue by andreasneumann** [opengisch/solocator#16]
Orthoimages and topographic maps.

Needs an update in the search and dataproduct service.

To be expected by the end of October.

**Comment by 3nids** [comment]
background layers are always added to the bottom of the layer tree

**Comment by andreasneumann** [comment]
This is now implemented in our test environment, but you would have to use

The filter "dataproduct" should be replaced by "foreground,background".

https://geo-t.so.ch/ as the hostname to use it for now, until it arrives in the integration and production environment. The same hostname should be used for both the search and the dataproduct service.

In the result json structure there is an attribute called "stacktype" which is either "foreground" or "background".

Example Query:
https://geo-t.so.ch/api/search/v2/?limit=10&filter=foreground,background&searchtext=Hintergrund
https://geo-t.so.ch/api/search/v2/?limit=10&filter=foreground,background&searchtext=Ortho

JSON Result for query "Hintergrundkarte":

```
{
	"results": [{
		"dataproduct": {
			"dataproduct_id": "ch.so.agi.hintergrundkarte_farbig",
			"stacktype": "background",
			"type": "facadelayer",
			"display": "Hintergrundkarte farbig",
			"dset_info": true
		}
	}, {
		"dataproduct": {
			"dataproduct_id": "ch.so.agi.hintergrundkarte_sw",
			"stacktype": "background",
			"type": "facadelayer",
			"display": "Hintergrundkarte schwarz-weiss",
			"dset_info": true
		}
	}, {
		"dataproduct": {
			"dataproduct_id": "ch.so.agi.hintergrundkarte_ortho",
			"stacktype": "background",
			"type": "facadelayer",
			"display": "Hintergrundkarte Orthofoto",
			"dset_info": true
		}
	}, {
		"dataproduct": {
			"sublayers": [{
				"dataproduct_id": "ch.so.afu.bodeninformationen.wald.verdichtungsempfindlichkeit",
				"type": "datasetview",
				"display": "Verdichtungsempfindlichkeit (Waldboden)",
				"dset_info": true
			}, {
				"dataproduct_id": "ch.so.afu.bodeninformationen.wald.hinweiskarte_bodenverdichtung",
				"type": "datasetview",
				"display": "Hinweiskarte Bodenverdichtung (Waldboden)",
				"dset_info": true
			}, {
				"dataproduct_id": "ch.so.afu.bodeninformationen.wald.charakter_wasserhaushalt",
				"type": "datasetview",
				"display": "Charakter des Wasserhaushaltes und Konsequenzen f\u00fcr die Durchwurzelung (Waldboden)",
				"dset_info": true
			}, {
				"dataproduct_id": "ch.so.afu.bodeninformationen.wald.bodenart_im_oberboden",
				"type": "datasetview",
				"display": "K\u00f6rnung (Waldboden)",
				"dset_info": true
			}, {
				"dataproduct_id": "ch.so.afu.bodeninformationen.wald.humusgehalt",
				"type": "datasetview",
				"display": "Humusgehalt (Waldboden)",
				"dset_info": true
			}, {
				"dataproduct_id": "ch.so.afu.bodeninformationen.wald.humusform",
				"type": "datasetview",
				"display": "Humusform (Waldboden)",
				"dset_info": true
			}, {
				"dataproduct_id": "ch.so.afu.bodeninformationen.wald.ph_oberboden",
				"type": "datasetview",
				"display": "pH im Oberboden (Waldboden)",
				"dset_info": true
			}, {
				"dataproduct_id": "ch.so.afu.bodeninformationen.wald.ph_unterboden",
				"type": "datasetview",
				"display": "pH im Unterboden (Waldboden)",
				"dset_info": true
			}, {
				"dataproduct_id": "ch.so.afu.bodeninformationen.wald.steingehalt_oberboden",
				"type": "datasetview",
				"display": "Steingehalt im Oberboden (Waldboden)",
				"dset_info": true
			}, {
				"dataproduct_id": "ch.so.afu.bodeninformationen.wald.steingehalt_unterboden",
				"type": "datasetview",
				"display": "Steingehalt im Unterboden (Waldboden)",
				"dset_info": true
			}],
			"display": "Bodeninformationen Wald",
			"dataproduct_id": "ch.so.afu.bodeninformation_wald",
			"stacktype": "foreground",
			"type": "layergroup",
			"dset_info": true
		}
	}, {
		"dataproduct": {
			"sublayers": [{
				"dataproduct_id": "ch.so.afu.bodeninformationen.lw.verdichtungsempfindlichkeit",
				"type": "datasetview",
				"display": "Verdichtungsempfindlichkeit (Landwirtschaftsboden)",
				"dset_info": true
			}, {
				"dataproduct_id": "ch.so.afu.bodeninformationen.lw.charakter_wasserhaushalt",
				"type": "datasetview",
				"display": "Charakter des Wasserhaushaltes (Landwirtschaftsboden)",
				"dset_info": true
			}, {
				"dataproduct_id": "ch.so.afu.bodeninformationen.lw.durchwurzelungstiefe",
				"type": "datasetview",
				"display": "Durchwurzelungstiefe und Speicherverm\u00f6gen f\u00fcr N\u00e4hrstoffe und Wasser (Landwirtschaftsboden)",
				"dset_info": true
			}, {
				"dataproduct_id": "ch.so.afu.bodeninformationen.lw.pflanzennutzbarer_wasservorrat",
				"type": "datasetview",
				"display": "maximaler pflanzennutzbarer Wasservorrat im Boden (Landwirtschaftsboden)",
				"dset_info": true
			}, {
				"dataproduct_id": "ch.so.afu.bodeninformationen.lw.bodenart_und_bearbeitbarkeit",
				"type": "datasetview",
				"display": "Bodenart und Bearbeitbarkeit (Landwirtschaftsboden)",
				"dset_info": true
			}, {
				"dataproduct_id": "ch.so.afu.bodeninformationen.lw.humusgehalt",
				"type": "datasetview",
				"display": "Humusgehalt (Landwirtschaftsboden)",
				"dset_info": true
			}, {
				"dataproduct_id": "ch.so.afu.bodeninformationen.lw.ph_oberboden",
				"type": "datasetview",
				"display": "pH im Oberboden (Landwirtschaftsboden)",
				"dset_info": true
			}, {
				"dataproduct_id": "ch.so.afu.bodeninformationen.lw.steingehalt_oberboden",
				"type": "datasetview",
				"display": "Steingehalt im Oberboden (Landwirtschaftsboden)",
				"dset_info": true
			}, {
				"dataproduct_id": "ch.so.afu.bodeninformationen.lw.nutzungsmoeglichkeiten",
				"type": "datasetview",
				"display": "Nutzungsm\u00f6glichkeiten, ohne Ber\u00fccksichtigung der Hangneigung (Landwirtschaftsboden)",
				"dset_info": true
			}, {
				"dataproduct_id": "ch.so.afu.bodeninformationen.lw.hangneigung",
				"type": "datasetview",
				"display": "Hangneigung und Einschr\u00e4nkungen im Ackerbau/Maschineneinsatz (Landwirtschaftsboden)",
				"dset_info": true
			}, {
				"dataproduct_id": "ch.so.afu.bodeninformationen.lw.erosionsgefahr",
				"type": "datasetview",
				"display": "Erosionsgefahr (Landwirtschaftsboden)",
				"dset_info": true
			}],
			"display": "Bodeninformationen Landwirtschaft",
			"dataproduct_id": "ch.so.afu.bodeninformation_landwirtschaft",
			"stacktype": "foreground",
			"type": "layergroup",
			"dset_info": true
		}
	}],
	"result_counts": []
}
```

**Comment by andreasneumann** [comment]
The configuration dialogue needs to reflect the changes:

![image](https://user-images.githubusercontent.com/884476/68196692-3da88e00-ffb9-11e9-8c44-711886965f85.png)

**Comment by 3nids** [comment]
@andreasneumann the new server apparently fails when calling with the same filter as before:

This fails: https://geo-t.so.ch/api/search/v2/?searchtext=rich&filter=foreground,background,ch.so.agi.gemeindegrenzen,ch.so.agi.av.gebaeudeadressen.gebaeudeeingaenge,ch.so.agi.av.bodenbedeckung,ch.so.agi.av.grundstuecke.projektierte,ch.so.agi.av.grundstuecke.rechtskraeftig,ch.so.agi.av.nomenklatur.flurnamen,ch.so.agi.av.nomenklatur.gelaendenamen&limit=20
While this is ok:
https://geo-t.so.ch/api/search/v2/?searchtext=rich&filter=foreground,background&limit=20

Did the layer names changed?

**Comment by andreasneumann** [comment]
You are right, the former fails. You can try with the "integration" server instead of the "test" server. There seems to be an issue with our test server.

In the URL, just replace the hostname geo-t with geo-i - so https://geo-i.so.ch/api/search/v2/?searchtext=rich&filter=foreground,background,ch.so.agi.gemeindegrenzen,ch.so.agi.av.gebaeudeadressen.gebaeudeeingaenge,ch.so.agi.av.bodenbedeckung,ch.so.agi.av.grundstuecke.projektierte,ch.so.agi.av.grundstuecke.rechtskraeftig,ch.so.agi.av.nomenklatur.flurnamen,ch.so.agi.av.nomenklatur.gelaendenamen&limit=20 should work.

**Comment by 3nids** [comment]
thanks for the prompt reply.

**Comment by 3nids** [comment]
Currenlty, inserting the layers appends at the currently selected entry in the legend.
If you want background layers to be under the others, for simple projects, they must be at the end of the layer tree. Is this what you'd like? 
Shall we put them into a folder or put them directly on the top level.
Have you considered the case of project with custom layer order?

**Comment by 3nids** [comment]
apart from last question this is implemented in 1.2.0

**Comment by andreasneumann** [comment]
Yes, we'd want to load the background layers at the very bottom on top level, not in a separate group.

About custom layer order: also in this case, it should appear at the very bottom of the layer list.
